src/System.Private.CoreLib/shared/System/Buffers/StandardFormat.cs

```diff
@@ -146,40 +146,47 @@ private static bool ParseHelper(ReadOnlySpan<char> format, out StandardFormat st
         /// </summary>
         public override string ToString()
         {
-            unsafe
+            Span<char> buffer = stackalloc char[FormatStringLength];
+            int charsWritten = Format(buffer);
+            return new string(buffer.Slice(0, charsWritten));
+        }
+
+        /// <summary>The length buffer needed by <see cref="Format"/>.</summary>
+        internal const int FormatStringLength = 4;
+
+        /// <summary>
+        /// Formats the format in classic .NET format.
+        /// </summary>
+        internal int Format(Span<char> destination)
+        {
+            Debug.Assert(destination.Length >= 4);
+
+            int count = 0;
+            char symbol = Symbol;
+            if (symbol != default)
             {
-                const int MaxLength = 4;
-                char* pBuffer = stackalloc char[MaxLength];
+                destination[count++] = symbol;
 
-                int dstIndex = 0;
-                char symbol = Symbol;
-                if (symbol != default)
+                byte precision = Precision;
+                if (precision != NoPrecision)
                 {
-                    pBuffer[dstIndex++] = symbol;
-
-                    byte precision = Precision;
-                    if (precision != NoPrecision)
+                    if (precision >= 100)
```

Comment:
Is this code path relevant for performance?, because you already have code for this in https://github.com/dotnet/corefx/blob/f5d31619f821e7b4a0bcf7f648fe1dc2e4e2f09f/src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs#L162 (just adapt for char) without all the % or use the DivRem implementation similar to the one from FormattingHelpers.
Or you could just get rid of the %10 in line 181, because precision can't be 100 or above in that code path (% 100 in 176), so division by 10 is also below 10.

Reply:
As we discussed offline, precision can never be > 99 and hence we can remove this code path (and maybe replace with `Debug.Assert`).